clang/lib/AST/CommentParser.cpp

```diff
@@ -149,6 +276,93 @@ class TextTokenRetokenizer {
     addToken();
   }
 
+  /// Extract a type argument
+  bool lexType(Token &Tok) {
+    if (isEnd())
+      return false;
+    Position SavedPos = Pos;
+    consumeWhitespace();
+    SmallString<32> NextToken;
+    SmallString<32> WordText;
+    const char *WordBegin = Pos.BufferPtr;
+    SourceLocation Loc = getSourceLocation();
+    StringRef ConstVal = StringRef("const");
+    StringRef PointerVal = StringRef("*");
+    StringRef ReferenceVal = StringRef("&");
+    bool ConstPointer = false;
+
+    while (!isEnd()) {
```

Comment:
Some comments wouldn't hurt...

Reply:
Added as part of the latest commit, along with some light refactoring to make it more readable. No doubt it is a complex piece of code